### ISSUE TITLE
Prometheus metrics hardcoded to start at port 8080

### ISSUE THREAD

**Issue by mmonaco** [GoogleCloudPlatform/gcs-fuse-csi-driver#392]
The metrics added in #310 listen separately at each mount, with a starting port of `8080`. We should be able to customize the start port, but I think it would need to come from the sidecar annotation (`gke-gcsfuse/prometheus-start-port: "9090"` for example).

With this, `prometheus-port` doesn't need to be a disallowed flag. Instead it would skip incrementing from the start port.

(Are there options to disable the metrics if users don't want them at all?)

**Comment by mmonaco** [comment]
Sure, that's a workaround, but taking `8080` by default is quite abrasive.

**Comment by sirohia** [comment]
I agree. The metrics collection port should be customizable. Hard coding to 8080 is very strict.

**Comment by hime** [comment]
Thank you all for your feedback. We have recently disabled metrics exporting by default in #375 to prevent these types of frictions when using the driver. Regarding the port, I agree that this is rather strict. Marking this issue to track this enhancement to increase flexibility with users.

**Comment by avkudryashov** [comment]
> We have recently disabled metrics exporting by default in https://github.com/GoogleCloudPlatform/gcs-fuse-csi-driver/pull/375

Thank you for the rapid fix. Can you also provide some information to the [Kubernetes Engine release notes](https://cloud.google.com/kubernetes-engine/docs/release-notes)? Since this is a breaking change, it would be good to know which GKE version includes the fixed driver version.

**Comment by hime** [comment]
fixed on #471
